### PR TITLE
🚨🚨🚨 Remove softmax for EfficientNetForImageClassification 🚨🚨🚨

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

 **\/!\ This is a breaking change /!\\** 

EfficientNet implementation erroneously added a softmax to the models logits in the classification head. This breaks with convention and [documentation](https://github.com/huggingface/transformers/blob/87c9d8a10f3935a46fc7b254d3bea97bca4bfcce/src/transformers/modeling_outputs.py#L1215) for the [model outputs](https://github.com/huggingface/transformers/blob/87c9d8a10f3935a46fc7b254d3bea97bca4bfcce/src/transformers/models/efficientnet/modeling_efficientnet.py#L623). 

This results in the very small probabilities seen in the hosted widgets on the checkpoints pages, as the logits effectively have softmax applied twice. 

image

Most of the efficientnet checkpoints were downloaded < 100 times in the past month. However, [efficientnet-b7](https://huggingface.co/google/efficientnet-b7) is more popular with a few thousand downloads. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by amyeroberts** [comment]
cc @rwightman Managed to track down the issue with the efficient net predictions on the hub

**Review by sgugger — APPROVED** [review]
Nice catch, and necessary breaking change, I agree!
